pkg/i2gw/provider.go

```diff
@@ -68,13 +70,19 @@ type CustomResourceReader interface {
 	ReadResourcesFromFile(ctx context.Context, filename string) error
 }
 
-// The ResourceConverter interface specifies all the implemented Gateway API resource
-// conversion functions.
+// The ResourceConverter interface specifies conversion functions from Ingress
+// and extensions into IR.
 type ResourceConverter interface {
+	// ToIR converts stored API entities associated with the Provider into IR.
+	ToIR() (IR, field.ErrorList)
+}
 
-	// ToGatewayAPIResources converts stored API entities associated
-	// with the Provider into GatewayResources.
-	ToGatewayAPI() (GatewayResources, field.ErrorList)
+// The ResourceConverter interface specifies conversion functions from IR
+// into Gateway and Gateway extensions.
+type ExtensionConverter interface {
+	// ToGatewayAPIAndExtensions converts stored IR with the Provider into
+	// Gateway API resources and extensions
+	ToGatewayAPIAndExtensions(IR) (GatewayResources, field.ErrorList)
```

Comment:
Do we really need to indicate extensions here? Also, how about two different functions on this interface, `ToGatewayAPI` and `ToGatewayExtensions` ? can also be `BuildGatewayAPIResources` & `BuildGatewayExtensions`

Reply:
I would say we we do `ToGatewayResources` here, but still having one struct and one function we writing all results to `GatewayResouces` struct.

Reply:
Why one function and not two? not sure I am following

Reply:
Discussed offline, we will still use one function for conversion so we output one GatewayResources directly.